code/modules/overmap/ships/computers/sensors.dm

```diff
@@ -128,17 +145,41 @@
 		for(var/obj/effect/overmap/visitable/identified_contact in contact_datums)
 			potential_contacts |= identified_contact
 
-		for(var/obj/effect/overmap/O in potential_contacts)
-			if(linked == O)
+		for(var/obj/effect/overmap/contact in potential_contacts)
+			if(linked == contact)
 				continue
-			if(!O.scannable)
+			if(!contact.scannable)
 				continue
-			var/bearing = round(90 - Atan2(O.x - linked.x, O.y - linked.y),5)
+
+			var/obj/effect/overmap/visitable/ship/landable/contact_landable = contact
+			var/landed = (istype(contact_landable) && contact_landable.status == SHIP_STATUS_LANDED)
+
+			var/contact_x = contact.x
+			var/contact_y = contact.y
+
+			var/obj/effect/overmap/visitable/ship/contact_ship = contact
+			if(istype(contact_ship))
+				contact_x += contact_ship.position[1] / 2.0
+				contact_y += contact_ship.position[2] / 2.0
+
+			var/bearing = round(90 - Atan2(contact_x - linked_x, contact_y - linked_y),5)
```

Comment:
This would probably enjoy to be a `#define` too, as it would be a valid bearing evaluation in any 2D space, and might be used in other places for something else down the line

Reply:
rewritten to a macro